src/api/blocks.py

```diff
@@ -0,0 +1,395 @@
+from __future__ import annotations
+
+from sys import getsizeof
+import re
+from typing import Dict, Iterable, List, Tuple, Union, overload
+
+
+class Block:
+    """
+    Class to handle data about various blockstates and allow for extra blocks to be created and interacted with.
+
+    .. important::
```

Comment:
The note here should be removed/changed because the get_block_instance function should not return a Block and the Block constructor should be used. (See other review comments below)

Reply:
I agree!